File: app/src/main/java/zeta/android/apps/presenter/NavigationPresenter.java
package zeta.android.apps.presenter;

import javax.annotation.ParametersAreNonnullByDefault;
import javax.inject.Inject;

import zeta.android.apps.BuildConfig;
import zeta.android.apps.rx.providers.RxSchedulerProvider;
import zeta.android.apps.ui.presentation.NavigationPresentation;

@ParametersAreNonnullByDefault
public class NavigationPresenter extends ZetaRxActivityLifeCyclePresenter<NavigationPresentation> {

    private NavigationPresentation mPresentation;

    @Inject
    public NavigationPresenter(RxSchedulerProvider schedulerProvider) {
        super(schedulerProvider);
    }

    @Override
    public void onCreate(NavigationPresentation navigationPresentation) {
        mPresentation = navigationPresentation;
        mPresentation.showDebugMenuItem(BuildConfig.DEBUG);
    }

    @Override
    public void onPostResume() {
        // no-op
    }

    @Override
    public void onPause() {
        // no-op
    }

    @Override
    public void onDestroy() {
        super.onDestroy();
        mPresentation = null;
    }

    public void onMenuItemHomeSelected() {
        mPresentation.showBaseScreen();
    }

    public void onMenuItemDebugSelected() {
        mPresentation.showDebugScreen();
    }
}

File: app/src/main/kotlin/zeta/android/apps/presenter/NavigationPresenter.kt
package zeta.android.apps.presenter

import javax.annotation.ParametersAreNonnullByDefault
import javax.inject.Inject

import zeta.android.apps.BuildConfig
import zeta.android.apps.presenter.ZetaRxActivityLifeCyclePresenter
import zeta.android.apps.rx.providers.RxSchedulerProvider
import zeta.android.apps.ui.presentation.NavigationPresentation

@ParametersAreNonnullByDefault
class NavigationPresenter @Inject
constructor(schedulerProvider: RxSchedulerProvider) : ZetaRxActivityLifeCyclePresenter<NavigationPresentation>(schedulerProvider) {

    private var mPresentation: NavigationPresentation? = null

    override fun onCreate(navigationPresentation: NavigationPresentation) {
        mPresentation = navigationPresentation
        mPresentation?.showDebugMenuItem(BuildConfig.DEBUG)
    }

    override fun onPostResume() {
        // no-op
    }

    override fun onPause() {
        // no-op
    }

    override fun onDestroy() {
        super.onDestroy()
        mPresentation = null
    }

    fun onMenuItemHomeSelected() {
        mPresentation?.showBaseScreen()
    }

    fun onMenuItemDebugSelected() {
        mPresentation?.showDebugScreen()
    }
}
